from Bio.Blast import NCBIWWW
import xml.etree.ElementTree as ET

# Homo sapiens brain-expressed HHCPA78 homolog VDUP1 (mRNA, complete cds) sequence with FASTA
# You can change the sequence input
sequence = ("GCTTAGTGTAACCAGCGGCGTATATTTTTTAGGCGCCTTTTCGAAAACCTAGTAGTTAATATTCATTTGTTTAAATCTTATTTTATTTTTAAGCTCAAACTGCTTAAGAATACCTTAATTCCTTAAAGTGAAATAATTTTTTGCAAAGGGGTTTCCTCGATTTGGAGCTTTTTTTTTCTTCCACCGTCATTTCTAACTCTTAAAACCAACTCAGTTCCATCATGGTGATGTTCAAGAAGATCAAGTCTTTTGAGGTGGTCTTTAACGACCCTGAAAAGGTGTACGGCAGTGGCGAGAGGGTGGCTGGCCGGGTGATAGTGGAGGTGTGTGAAGTTACTCGTGTCAAAGCCGTTAGGATCCTGGCTTGCGGAGTGGCTAAAGTGCTTTGGATGCAGGGATCCCAGCAGTGCAAACAGACTTCGGAGTACCTGCGCTATGAAGACACGCTTCTTCTGGAAGACCAGCCAACAGGTGAGAATGAGATGGTGATCATGAGACCTGGAAACAAATATGAGTACAAGTTCGGCTTTGAGCTTCCTCAGGGGCCTCTGGGAACATCCTTCAAAGGAAAATATGGGTGTGTAGACTACTGGGTGAAGGCTTTTCTTGACCGCCCGAGCCAGCCAACTCAAGAGACAAAGAAAAACTTTGAAGTAGTGGATCTGGTGGATGTCAATACCCCTGATTTAATGGCACCTGTGTCTGCTAAAAAAGAAAAGAAAGTTTCCTGCATGTTCATTCCTGATGGGCGGGTGTCTGTCTCTGCTCGAATTGACAGAAAAGGATTCTGTGAAGGTGATGAGATTTCCATCCATGCTGACTTTGAGAATACATGTTCCCGAATTGTGGTCCCCAAAGCTGCCATTGTGGCCCGCCACACTTACCTTGCCAATGGCCAGACCAAGGTGCTGACTCAGAAGTTGTCATCAGTCAGAGGCAATCATATTATCTCAGGGACATGCGCATCATGGCGTGGCAAGAGCCTTCGGGTTCAGAAGATCAGGCCTTCTATCCTGGGCTGCAACATCCTTCGAGTTGAATATTCCTTACTGATCTATGTTAGCGTTCCTGGATCCAAGAAGGTCATCCTTGACCTGCCCCTGGTAATTGGCAGCAGATCAGGTCTAAGCAGCAGAACATCCAGCATGGCCAGCCGAACCAGCTCTGAGATGAGTTGGGTAGATCTGAACATCCCTGATACCCCAGAAGCTCCTCCCTGCTATATGGATGTCATTCCTGAAGATCACCGATTGGAGAGCCCAACAACTCCTCTGCTAGATGACATGGATGGCTCTCAAGACAGCCCTATCTTTATGTATGCCCCTGAGTTCAAGTTCATGCCACCACCGACTTATACTGAGGTGGATCCCTGCATCCTCAACAACAATGTGCAGTGAGCATGTGGAAGAAAAGAAGCAGCTTTACCTACTTGTTTCTTTTTGTCTCTCTTCCTGGACACTCACTTTTTCAGAGACTCAACAGTCTCGTCAATGGAGTGTGGGTCCACCTTAGCCTCTGACTTCCTAATGTAGGAGGTGGTCAGCAGGCAATCTCCTGGGCCTTAAAGGATGCGGACTCATCCTCAGCCAGCGCCCATGTTGTGATACAGGGGTGTTTGTTGGATGGGTTTAAAAATAACTAGAAAAACTCAGGCCCATCCATTTTCTCAGATCTCCTTGAAAATTGAGGCCTTTTCGATAGTTTCGGGTCAGGTAAAAATGGCCTCCTGGCGTAAGCTTTTCAAGGTTTTTTGGAGGCTTTTTGTAAATTGTGATAGGAACTTTGGACCTTGAACTTACGTATCATGTGGAGAAGAGCCAATTTAACAAACTAGGAAGATGAAAAGGGAAATTGTGGCCAAAACTTTGGGAAAAGGAGGTTCTTAAAATCAGTGTTTCCCCTTTGTGCACTTGTAGAAAAAAAAGAAAAACCTTCTAGAGCTGATTTGATGGACAATGGAGAGAGCTTTCCCTGTGATTATAAAAAAGGAAGCTAGCTGCTCTACGGTCATCTTTGCTTAGAGTATACTTTAACCTGGCTTTTAAAGCAGTAGTAACTGCCCCACCAAAGGTCTTAAAAGCCATTTTTGGAGCCTATTGCACTGTGTTCTCCTACTGCAAATATTTTCATATGGGAGGATGGTTTTCTCTTCATGTAAGTCCTTGGAATTGATTCTAAGGTGATGTTCTTAGCACTTTAATTCCTGTCAAATTTTTTGTTCTCCCCTTCTGCCATCTTAAATGTAAGCTGAAACTGGTCTACTGTGTCTCTAGGGTTAAGCCAAAAGACAAAAAAAATTTTACTACTTTTGAGATTGCCCCAATGTACAGAATTATATAATTCTAACGCTTAAATCATGTGAAAGGGTTGCTGCTGTCAGCCTTGCCCACTGTGACTTCAAACCCAAGGAGGAACTCTTGATCAAGATGCCCAACCCTGTGATCAGAACCTCCAAATACTGCCATGAGAAACTAGAGGGCAGGTGTTCATAAAAGCCCTTTGAACCCCCTTCCTGCCCTGTGTTAGGAGATAGGGATATTGGCCCCTCACTGCAGCTGCCAGCACTTGGTCAGTCACTCTCAGCCATAGCACTTTGTTCACTGTCCTGTGTCAGAGCACTGAGCTCCACCCTTTTCTGAGAGTTATTACAGCCAGAAAGTGTGGGCTGAAGATGGTTGGTTTCATGTGGGGGTATTATGTACCC")

# NCBIWWW.gblast(program, database, sequence)
result_handle = NCBIWWW.qblast("blastn", "nt", sequence)

"""
Types of programs and databases can be see through the web 
@https://blast.ncbi.nlm.nih.gov/blast/Blast.cgi?PROGRAM=tblastn&PAGE_TYPE=BlastSearch&LINK_LOC=blasthome
programs: blastn, blastp, etc.
database: the code can usually be seen in brackets in the database dropdown
"""
# Handling the results
blast_results = result_handle.read()
root = ET.fromstring(blast_results)

# convert to HTML
html_output = "<html><body><h1>BLAST Results</h1><ul>"
for alignment in root.iter("Hit"):
    hit_def = alignment.find("Hit_def").text
    html_output += f"<li>{hit_def}</li>"
html_output += "</ul></body></html>"
html_filename = "blast_results.html"
with open(html_filename, "w") as file:
    file.write(html_output)

print(f"saved to {html_filename}.")
